presto-orc/src/main/java/com/facebook/presto/orc/reader/TimestampStreamReader.java

```diff
@@ -229,6 +229,9 @@ public static long decodeTimestamp(long seconds, long serializedNanos, long base
     {
         long millis = (seconds + baseTimestampInSeconds) * MILLIS_PER_SECOND;
         long nanos = parseNanos(serializedNanos);
+        if (nanos > 999999999 || nanos < 0) {
+            throw new IllegalArgumentException("nanos > 999999999 or < 0");
```

Comment:
`nano field of an encoded timestamp in ORC must be between 0 and 999999999 inclusive, got XXX`

Reply:
@haozhun : This exception message is copied from `java.sql.Timestamp`: http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/6-b14/java/sql/Timestamp.java#90 . But agree it's quite vague :)